.github/workflows/builds.yaml

```diff
@@ -0,0 +1,43 @@
+name: Build Tests
+
+on:
+  schedule:
+    - cron: "0 2 * * *"
+  workflow_dispatch
+
+jobs:
+  build:
+    runs-on: [self-hosted]
+    strategy:
+      fail-fast: false
+    compiler:
+      - gcc/9.5.0
+      - gcc/10.5.0
+      - gcc/12.3.0
+      - gcc/13.2.0
+      - clang/12.0.1
+      - clang/14.0.6
+      - clang/16.0.6
+      - clang/17.0.6
+      - intel/oneAPI-2021.4
```

Comment:
this is still not correct
```
strategy:
  fail-fast: false
  matrix:
    compiler:
      - gcc/...
      - ...
```
See [doco](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy)